### ISSUE TITLE
Issue for the code

### ISSUE THREAD

**Issue by YiDongOuYang** [maple-research-lab/LLaDOU#3]
Dear authors, Thank you very much for your great work. I have a question regarding this [line](https://github.com/maple-research-lab/LLaDOU/blob/97ff0cc46d6a11110e281f9107dc1f59f0ec5eed/networks/lladou_v0.py#L305). Since old_log_probs = log_probs.detach(),
coeff_1 = 1 under any conditions right?

**Comment by Faustrazor** [comment]
在一般的GRPO中，对于一个problem，roll_out出来8个答案，这8个答案会重复多次重复利用去更新模型参数。在8个答案重复利用的过程中，会有一个old_log_probs，在roll_out之后就确定了下来，每次重复利用更新参数后，会得到一个log_probs，由当前模型参数计算得到。但如果8个答案不进行重复利用，仅更新一次参数梯度，那么old_log_probs和log_probs就是相等的，只是一个记录了梯度用于后续更新参数，另一个不记录梯度。

**Comment by YiDongOuYang** [comment]
> 在一般的GRPO中，对于一个problem，roll_out出来8个答案，这8个答案会重复多次重复利用去更新模型参数。在8个答案重复利用的过程中，会有一个old_log_probs，在roll_out之后就确定了下来，每次重复利用更新参数后，会得到一个log_probs，由当前模型参数计算得到。但如果8个答案不进行重复利用，仅更新一次参数梯度，那么old_log_probs和log_probs就是相等的，只是一个记录了梯度用于后续更新参数，另一个不记录梯度。

谢谢你的讨论 但是从代码逻辑来看 old_log_probs并不是在roll_out中进行计算的 这意味着old_log_probs仍然使用的是最新的模型参数 （即与log_probs相同的模型参数）如果我对[logprob_loss](https://github.com/maple-research-lab/LLaDOU/blob/97ff0cc46d6a11110e281f9107dc1f59f0ec5eed/networks/lladou_v0.py#L226C5-L226C17)这个函数理解正确的话

**Comment by Faustrazor** [comment]
是最新的模型参数，old_log_probs在roll_out结束以后就确定了，而log_probs会一直更新，但第一次更新的时候他们是相同的。如果这组样本仅用一次的话，old_log_probs计算和log_probs没有本质区别

**Comment by YiDongOuYang** [comment]
> 是最新的模型参数，old_log_probs在roll_out结束以后就确定了，而log_probs会一直更新，但第一次更新的时候他们是相同的。如果这组样本仅用一次的话，old_log_probs计算和log_probs没有本质区别

请问你说“old_log_probs在roll_out结束以后就确定了，而log_probs会一直更新”是对这个代码的说明吗？我知道这是GRPO实现的逻辑，但我认为在这个代码中即使infer_kwargs['repeat_times']>1,old_log_probs和log_probs的值依然相同，导致coeff_1=coeff_2=1

**Comment by Faustrazor** [comment]
是的，但是infer_kwargs['repeat_times']这个参数，我认为作者并不是想表达对采样出来的样本重复利用的次数，这个参数我觉得和num_generations没有区别
